OpenRA.Game/Network/Order.cs

```diff
@@ -156,7 +156,18 @@ public static Order Deserialize(World world, BinaryReader r)
 									else
 									{
 										var pos = new WPos(r.ReadInt32(), r.ReadInt32(), r.ReadInt32());
-										target = Target.FromPos(pos);
+
+										var numberOfTerrainPositions = r.ReadInt16();
+										if (numberOfTerrainPositions == -1)
+											target = Target.FromPos(pos);
```

Comment:
no further bound checks.

Reply:
What bound checks do you mean? We don't read more from the reader here.

Reply:
values like -2 and Int.MaxValue could cause the game to crash out of memory. any invalid serialized protocol order.
but i am not suggested to check this. just that the serialization of order could be abused.

Reply:
I agree, we shouldn't crash here

Reply:
I made #21036